Cannot compile: "Templates cannot be declared to have 'C' Linkage"

When output when building:

```1>E:\1_Development\FreeTime\BasicChatApp\libs\Nuklear\include\nuklear.h(5707,1): error C2894: templates cannot be declared to have 'C' linkage
1>E:\1_Development\FreeTime\BasicChatApp\libs\Nuklear\include\nuklear.h(5708,1): error C2894: templates cannot be declared to have 'C' linkage
1>E:\1_Development\FreeTime\BasicChatApp\libs\Nuklear\include\nuklear.h(5709,1): error C2894: templates cannot be declared to have 'C' linkage
1>E:\1_Development\FreeTime\BasicChatApp\libs\Nuklear\include\nuklear.h(5710,1): error C2894: templates cannot be declared to have 'C' linkage
1>E:\1_Development\FreeTime\BasicChatApp\libs\Nuklear\include\nuklear.h(7382,10): error C4996: 'fopen': This function or variable may be unsafe. Consider using fopen_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.
1>Window.cpp
1>E:\1_Development\FreeTime\BasicChatApp\libs\Nuklear\include\nuklear.h(5707,1): error C2894: templates cannot be declared to have 'C' linkage
1>E:\1_Development\FreeTime\BasicChatApp\libs\Nuklear\include\nuklear.h(5708,1): error C2894: templates cannot be declared to have 'C' linkage
1>E:\1_Development\FreeTime\BasicChatApp\libs\Nuklear\include\nuklear.h(5709,1): error C2894: templates cannot be declared to have 'C' linkage
1>E:\1_Development\FreeTime\BasicChatApp\libs\Nuklear\include\nuklear.h(5710,1): error C2894: templates cannot be declared to have 'C' linkage
1>E:\1_Development\FreeTime\BasicChatApp\libs\Nuklear\include\nuklear.h(7382,10): error C4996: 'fopen': This function or variable may be unsafe. Consider using fopen_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.
1>Done building project "EchoClient.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

The error points me to:

```cpp
#ifdef __cplusplus
template<typename T> struct nk_alignof;
template<typename T, int size_diff> struct nk_helper{enum {value = size_diff};};
template<typename T> struct nk_helper<T,0>{enum {value = nk_alignof<T>::value};};
template<typename T> struct nk_alignof{struct Big {T x; char c;}; enum {
    diff = sizeof(Big) - sizeof(T), value = nk_helper<Big, diff>::value};};
#define NK_ALIGNOF(t) (nk_alignof<t>::value)
#else
#define NK_ALIGNOF(t) NK_OFFSETOF(struct {char c; t _h;}, _h)
#endif```

--- a/nuklear.h
+++ b/nuklear.h
@@ -5704,6 +5704,10 @@ struct nk_context {
 #endif
 
 #ifdef __cplusplus
+}
+#endif
+
+#ifdef __cplusplus
 template<typename T> struct nk_alignof;
 template<typename T, int size_diff> struct nk_helper{enum {value = size_diff};};
 template<typename T> struct nk_helper<T,0>{enum {value = nk_alignof<T>::value};};
@@ -5717,9 +5721,6 @@ template<typename T> struct nk_alignof{struct Big {T x; char c;}; enum {
 #define NK_CONTAINER_OF(ptr,type,member)\
     (type*)((void*)((char*)(1 ? (ptr): &((type*)0)->member) - NK_OFFSETOF(type, member)))
 
-#ifdef __cplusplus
-}
-#endif
 
 
 #endif /* NK_NUKLEAR_H_ */
@@ -29536,6 +29537,7 @@ nk_tooltipfv(struct nk_context *ctx, const char *fmt, va_list args)
 ///    - [yy]: Minor version with non-breaking API and library changes
 ///    - [zz]: Bug fix version with no direct changes to API
 ///
+/// - 2021/09/15 (4.08.3) - Fix "Templates cannot be declared to have 'C' Linkage"
 /// - 2021/09/08 (4.08.2) - Fix warnings in C89 builds
 /// - 2021/09/08 (4.08.1) - Use compiler builtins for NK_OFFSETOF when possible
 /// - 2021/08/17 (4.08.0) - Implemented 9-slice scaling support for widget styles

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "nuklear",
-  "version": "4.08.2",
+  "version": "4.08.3",
   "repo": "Immediate-Mode-UI/Nuklear",
   "description": "A small ANSI C gui toolkit",
   "keywords": ["gl", "ui", "toolkit"],

--- a/src/CHANGELOG
+++ b/src/CHANGELOG
@@ -8,6 +8,7 @@
 ///    - [yy]: Minor version with non-breaking API and library changes
 ///    - [zz]: Bug fix version with no direct changes to API
 ///
+/// - 2021/09/15 (4.08.3) - Fix "Templates cannot be declared to have 'C' Linkage"
 /// - 2021/09/08 (4.08.2) - Fix warnings in C89 builds
 /// - 2021/09/08 (4.08.1) - Use compiler builtins for NK_OFFSETOF when possible
 /// - 2021/08/17 (4.08.0) - Implemented 9-slice scaling support for widget styles

--- a/src/nuklear.h
+++ b/src/nuklear.h
@@ -5483,6 +5483,10 @@ struct nk_context {
 #endif
 
 #ifdef __cplusplus
+}
+#endif
+
+#ifdef __cplusplus
 template<typename T> struct nk_alignof;
 template<typename T, int size_diff> struct nk_helper{enum {value = size_diff};};
 template<typename T> struct nk_helper<T,0>{enum {value = nk_alignof<T>::value};};
@@ -5496,9 +5500,6 @@ template<typename T> struct nk_alignof{struct Big {T x; char c;}; enum {
 #define NK_CONTAINER_OF(ptr,type,member)\
     (type*)((void*)((char*)(1 ? (ptr): &((type*)0)->member) - NK_OFFSETOF(type, member)))
 
-#ifdef __cplusplus
-}
-#endif
 
 
 #endif /* NK_NUKLEAR_H_ */

